### ISSUE TITLE
This script appears to brick routers if it fails

### ISSUE THREAD

**Issue by paidforby** [sudomesh/tp-flasher#1]
I can't seem, to decipher how this script actually is supposed to work. I've tried following the given instructions after, running ```sudo apt install nodejs-legacy``` which was needed to even run it. I attempted to run the script as intended (also discovered a typo, will commit once i make more worthwhile additions) I seemed to have "soft" bricked two TP-Link WDR4300 routers. This soft bricking looks like all the front lights blinking every few seconds. I can no longer get into the state as described by @redconfetti in [this blog post](http://www.rubycoloredglasses.com/2016/04/tp-link-wdr4300-recovery/) so I can put the factory firmware back on it.

**Comment by Juul** [comment]
Well crap. I can have a look at them. Maybe tomorrow (technically today)
evening. Might need to hook up to serial console. I've flashed plenty of
these but it's been a while. Maybe the firmware/bootloader changed.

**Comment by paidforby** [comment]
I won't be around tonight, but I can drop them off in the cage this morning if you want to look at them, signal me if you have trouble finding them.

But certainly, it seems highly likely that something may have changed in the firmware since you last flashed one, or maybe I just followed the steps incorrectly, its really difficult to tell.

Also here is the (one of) errors I'm receiving when I try to run flasher.js while it is in the bricked state I described,
```
{ address: '192.168.0.66',
  netmask: '255.255.255.0',
  family: 'IPv4',
  mac: '0c:54:a5:1d:18:4e',
  internal: false }
Starting tftp server on 192.168.0.66 port 69
Ensure you are connected with an ethernet cable to one of the LAN ports on the router
Power off the router. Then hold down the reset button on the back of the router, turn it on, wait 10 seconds, then release the reset button and wait for the flashing to begin.
Server errorL { [Error: bind EADDRINUSE 192.168.0.66:69]
  code: 'EADDRINUSE',
  errno: 'EADDRINUSE',
  syscall: 'bind',
  address: '192.168.0.66',
  port: 69 }
```

**Comment by Juul** [comment]
Did you drop them off? If so I can stop by this evening.

**Comment by paidforby** [comment]
yes, there are two TP-Link WDR4300s (one is in its box) right inside the sudomesh cage, you shouldn't be able to miss them. Both of them are in the same, apparently bricked state. Thanks for taking a look at them.

**Comment by Juul** [comment]
Alright I fixed the bugs i tp-flasher and used it to flash theone with the
CCL label on it with the latest firmware from here:

https://builds.sudomesh.org/builds/sudowrt/fledgling/0.2.2/ar71xx/openwrt-ar71xx-generic-tl-wdr4300-v1-squashfs-sysupgrade.bin

The other one accepted the firmware but then just went into a state where
the link was up but it didn't respond to anything. Rebooting it took it
back to the blinking behavior as if the firmware was never flashed.

**Comment by Juul** [comment]
Hm I noticed that there are other variants of the firmware for slightly
different versions of the wdr4300. Maybe worth a shot?

**Comment by paidforby** [comment]
Awesome, thanks for working on this, I'll try reflashing the CCL labeled one this afternoon, update the README and wiki with any additional instructions and close this issue accordingly.

I can also take a look at that bricked one and maybe try a variant of the firmware.

**Comment by Juul** [comment]
Maybe try flashing with the official firmware? It could be that it's some half-assed protection.

Also, the flasher has a weird "feature" where you can't start it when the ethernet link is down, even though the link won't come up until you boot the router. It should probably just start regardless of whether the link is up.
